### ISSUE TITLE
Tags/categories for tilesets

### ISSUE THREAD

**Issue by Sipondo** [deepnight/ldtk#578]
**Problem description**
Tilesets are at the core of LDTK. Not only do they declare the tiles the world is made out of, they are used for more and more features, e.g. Enums and editor sprites for Entities.

Tilesets are currently stored in what is essentially a big dictionary with the tilesets' names as keys. Regardless whether the user should use one big atlas for all their graphics or multiple tilesets, it would be great to improve the flexibility of LDTK in this department. After all, people make a range of different games in LDTK, and very large games may still have to use a plethora of tilesets even when using big atlasses.

**Suggestion**
My suggestion is to implement a tagging/category system for tilesets that is similar to what we currently have for entities. This prevents having to write a radical solution, as the design can simply be copied from the way entities are categorised. Moreover, the tag/category system is highly optional and should improve compatibility with large projects without negatively impacting small projects.

**Comment by deepnight** [comment]
This would be cool to have that in Enums as well 🤔

**Comment by Sipondo** [comment]
Definitely! I'd it suits to be a core LDTK design pattern that pops up everywhere to organise stuff. :)

**Comment by deepnight** [comment]
Done in 1.0
